doc/port_auto_neg/port-auto-negotiation-design.md

```diff
@@ -390,15 +404,19 @@ Here is the table to map the fields and SAI attributes:
 #### State DB Enhancements
 
 To support validate interface speed on CLI side, a new field **supported_speeds** will be added to **PORT_TABLE**.
+To support checking remote advertised speeds, a new field **rmt_adv_speeds** will be added to **PORT_TABLE**.
 
 	; Defines information for port state
 	key                     = PORT_TABLE:port_name           ; state of the port
 	; field                 = value
     ...
 	supported_speeds        = STRING                         ; supported speed list
-  speed                   = STRING                         ; operational speed
+	speed                   = STRING                         ; operational speed
+	rmt_adv_speeds          = STRING                         ; advertised speed list of the remote
```

Comment:
The remote advertised speed might impact output of existing CLI. This usecase was not discussed in the meetings. Do we need this field?

Reply:
It's okay to not to show remote advertised speed in the CLI, it's only a hint to help users identify the cause of a link failure

Reply:
Just for your reference, while it's unexpected, the link could get up for some when local is with AutoNeg and remote is in fixed speed. For example, a 100M link could be up when local is with AutoNeg=OFF and remote is with AutoNeg=ON, if the fallback mode on the PHY is 100M by default.
Actually, we did hit this issue a few days ago, and here is the summary, and having the remote advertisement displayed in CLI could help users identify this issue, if the link is up when AutoNeg=ON, and the remote advertisement is NONE/EMPTY, it's this case.

e.g. The 100M link is somehow coming up unexpectedly.

DUT A                     DUT B
-----------------      ------------------------------------
100M, AN=OFF     AN=ON, Adv.Speed=1000,100,10

Unfortunately, although the link is up unexpectedly, the duplex mode is wrong, and caused traffics.

i.e.
DUT A is at Full-Duplex, while DUT B is at Half-Duplex.

Reply:
@Junchao-Mellanox Can you please provide feedback on modifying existing CLI's impact?

Reply:
Just for your reference, the CLI output is updated as follows, the column is increated from 90 to 108

**Original**
```
  Interface    Auto-Neg Mode    Speed    Adv Speeds    Type    Adv Types    Oper    Admin
-----------  ---------------  -------  ------------  ------  -----------  ------  -------
  Ethernet0          enabled      25G       10G,50G     CR4      CR4,CR2    down       up
 Ethernet32         disabled      40G           all     N/A          all      up       up
Ethernet112              N/A      40G           N/A     N/A          N/A      up       up
Ethernet116              N/A      40G           N/A     N/A          N/A      up       up
Ethernet120              N/A      40G           N/A     N/A          N/A      up       up
Ethernet124              N/A      40G           N/A     N/A          N/A      up       up
```

**Modified**

```
  Interface    Auto-Neg Mode    Speed    Adv Speeds    Rmt Adv Speeds    Type    Adv Types    Oper    Admin
-----------  ---------------  -------  ------------  ----------------  ------  -----------  ------  -------
  Ethernet0          enabled      25G       10G,50G               40G     CR4      CR4,CR2    down       up
 Ethernet32         disabled      40G           all               N/A     N/A          all      up       up
Ethernet112              N/A      40G           N/A               N/A     N/A          N/A      up       up
Ethernet116              N/A      40G           N/A               N/A     N/A          N/A      up       up
Ethernet120              N/A      40G           N/A               N/A     N/A          N/A      up       up
Ethernet124              N/A      40G           N/A               N/A     N/A          N/A      up       up
```